src/charm.py

```diff
@@ -1063,7 +1063,11 @@ def _on_start(self, event: StartEvent) -> None:
         if not self._can_start(event):
             return
 
-        postgres_password = self._get_password()
+        try:
+            postgres_password = self._get_password()
+        except ModelError:
+            logger.debug("_on_start: secrets not yet available")
+            postgres_password = None
```

Comment:
Sometimes fails in CI. Eg: https://github.com/canonical/pgbouncer-operator/actions/runs/10086523896/job/27890280183#step:29:1933

Reply:
@juditnovak please consider to report Juju if it is their issue OR fix our lib (no rush!).
In my mind, there should be no moment when secrets are not-yet available for the charm.
Correct me if I am wrong. Tnx!